sequencer/src/block.rs

```diff
@@ -308,43 +298,25 @@ impl Header {
             transactions_root,
             fee_merkle_tree_root,
             block_merkle_tree_root,
-            validated_state,
         }
     }
 }
 
-fn _validate_proposal(parent: &Header, proposal: &Header) -> anyhow::Result<BlockMerkleTree> {
```

Comment:
Removed in favor of `ValidatedState::validate_proposal()`. The signature has changed there somewhat so please verify this is heading in the right direction.  See below comment.